### PR TITLE
Small refactoring and fixes to `dinterpret.d`

### PR DESCRIPTION
**Pull request by Geod24**
I have a fix for an issue that is blocking #11000 but did a few refactorings before, so making a separate PR for them.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/dinterpret.d [hunk 1]
```diff
@@ -2163,37 +2159,39 @@ public:
...
-                    result = interpret(pue, ev, istate, goal);
+                    if (!v.isCTFE() && v.isDataseg())
+                        e.error("static variable `%s` cannot be read at compile time", v.toChars());
+                    else // CTFE initiated from inside a function
+                        e.error("variable `%s` cannot be read at compile time", v.toChars());
+                    result = CTFEExp.cantexp;
                     return;
                 }
+
+                if (v.storage_class & (STC.out_ | STC.ref_))
```
Comment (UplinkCoder) [inline src/dmd/dinterpret.d:2182]: Please don't mix functional changes with refactoring.

Comment (NilsLankila) [reply src/dmd/dinterpret.d:2182]: stoped the merge process until this review comment is resolved.

Comment (Geod24) [reply src/dmd/dinterpret.d:2182]: It is not a functional change. If you look at the commit in isolation, it explains that this condition is redundant.

Comment (UplinkCoder) [reply src/dmd/dinterpret.d:2182]: fair enough

**Review by UplinkCoder — APPROVED** [review]
Good to go.

**Comment by UplinkCoder** [comment]
Altough I don't see why the code has to touched.

It was fine as it was.

**Comment by NilsLankila** [comment]
yes okay, I just wanted to be sure that there was no problem.

**Review by ghost — APPROVED** [review]

**Comment by Geod24** [comment]
> Although I don't see why the code has to be touched.

Well there's 4 commits. I think we can all agree the `printf` fixes are no-brainer. I also hope we can agree that using more idiomatic D in the compiler, like is done with the `enum`, is beneficial for readability.

That leaves 2 commits shuffling a function. The reason the first commit came was to reduce the scope of the variable and remove the redundant `v` check. Not a big deal, but I originally had a lot more code around this function. Later on I realized I could also remove the redundant `hasValue` check, which is a bit more beneficial than removing a few duplicated `null` check. So if it wasn't for this, I would have seriously considered limiting this PR to two commits.

Anyway, the actual bug fix is coming now.

**Pull request merged**